### ISSUE TITLE
Retouch module does not show/apply correction

### ISSUE THREAD

**Issue by antoniopiazza** [aurelienpierreeng/ansel#310]
## Description of the bug

The "Retouch" module does not display or apply corrections.

The circular shape clone tool fails to apply the correction. When attempting the same correction in Darktable, the results are successful. Importing the XMP file from Darktable into Ansel results in the correction being successfully applied.

![image_2](https://github.com/aurelienpierreeng/ansel/assets/11981243/e8e527af-00ca-41a9-9b44-83cb7f1cd3f3)
![image_1](https://github.com/aurelienpierreeng/ansel/assets/11981243/b4139b73-c923-4de5-9d99-568ab23ce6fd)

**Comment by vtyrtov** [comment]
Yes, after "Composition guides: general rework" commit, retouch apply only when the grid is enabled.

**Comment by antoniopiazza** [comment]
"I updated this morning from Git. Issue resolved."

**Comment by vtyrtov** [comment]
The problem is reproduced starting from Ansel-ba4dc39-x86_64.AppImage by the current build.

The problem is relevant the first time shape is added to the module. When you reopen the image or add shapes, the problem goes away.

The console displays:
```
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1704386040 from array
rt_process_forms: missing form=1704386040 from array
```

**Comment by pjhyor7** [comment]
Im having this problem too. It seems that sometimes when you add a mask you can't alter its opacity either.

**Comment by NKlug** [comment]
I'm having the same problem with build Ansel-0.0.0+729~ge2c4a0a-x86_64

**Comment by Annabel2** [comment]
I too have this problem with Ansel-0.0.0+748~g61eb388-x86_64.AppImage.  Here are some lines from the Terminal window -
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397118 from array
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397118 from array
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397118 from array
rt_process_forms: missing form=1709397118 from array
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397118 from array
rt_process_forms: missing form=1709397118 from array
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397165 from array
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397165 from array
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397165 from array
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397213 from array
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397213 from array
mask 1709397165 from selection, parent: 1709397168
[_dev_add_history_item_ext] invalidating history
[_dev_add_history_item_ext] invalidating history
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397213 from array
rt_process_forms: missing form=1709397165 from array
rt_process_forms: missing form=1709397213 from array

**Comment by aurelienpierre** [comment]
Known, those are side-effects on the current pixel cache clean-up. Cache consistency is fixed from the most generic modules to the most peculiar, and this one is very peculiar.
